### PR TITLE
Compare function new api

### PR DESCRIPTION
**Pull request by pelszkow**
Ticket 47184

### CODE REVIEW

**Comment by pelszkow** [comment]
@GlebKazantaev is this new API for function comparison OK?

File: inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.hpp [hunk 1]
```diff
@@ -18,43 +18,77 @@
...
+        PRECISIONS = 1 << 3,
+        ATTRIBUTES = 1 << 4,
+    };
+
+    struct Result {
+        bool valid;
+        std::string message;
+    };
+
+    static constexpr FunctionsComparator no_default() noexcept {
```
Comment (ilyachur) [inline inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.hpp:37]: What do names `no_default` and `with_default` means?

File: inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.hpp [hunk 2]
```diff
@@ -18,43 +18,77 @@
...
+private:
+    constexpr explicit FunctionsComparator(CmpValues f) noexcept : m_comparition_flags(f) {}
+    CmpValues m_comparition_flags;
+};
+
+///
+/// \deprecated
+/// \brief compare_functions is obsolete function use FunctionComparator instead.
+///
+inline std::pair<bool, std::string> compare_functions(
```
Comment (ilyachur) [inline inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.hpp:69]: Why cannot we remove this method if we implemented the new approach?

File: inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.hpp (hunk 1 above)
Comment (pelszkow) [reply inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.hpp:37]: `with_default` cover defaults from `compare_function` and for `no_default` all conditional check are disabled.

File: inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.hpp (hunk 2 above)
Comment (pelszkow) [reply inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.hpp:69]: I assume it should be done in the next step.

**Review by GlebKazantaev — APPROVED** [review]

File: inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.hpp (hunk 2 above)
Comment (ilyachur) [reply inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.hpp:69]: Ok, let's plan and remove this functionality.

**Review by ilyachur — APPROVED** [review]

**Pull request merged**